spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/RowDataReader.java

```diff
@@ -46,25 +47,34 @@
 import org.apache.iceberg.types.TypeUtil;
 import org.apache.spark.rdd.InputFileBlockHolder;
 import org.apache.spark.sql.catalyst.InternalRow;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 class RowDataReader extends BaseDataReader<InternalRow> {
-
+  private static final Logger LOG = LoggerFactory.getLogger(RowDataReader.class);
   private final Schema tableSchema;
   private final Schema expectedSchema;
   private final String nameMapping;
   private final boolean caseSensitive;
+  private final DeleteCounter counter;
 
   RowDataReader(CombinedScanTask task, Table table, Schema expectedSchema, boolean caseSensitive) {
     super(table, task);
     this.tableSchema = table.schema();
     this.expectedSchema = expectedSchema;
     this.nameMapping = table.properties().get(TableProperties.DEFAULT_NAME_MAPPING);
     this.caseSensitive = caseSensitive;
+    this.counter = new DeleteCounter();
+  }
+
+  protected DeleteCounter counter() {
+    return counter;
   }
 
   @Override
   CloseableIterator<InternalRow> open(FileScanTask task) {
-    SparkDeleteFilter deletes = new SparkDeleteFilter(task, tableSchema, expectedSchema);
+    LOG.debug("calling open, constructing SparkDeleteFilter ...");
```

Comment:
Capital here, but i'm not sure this is a helpful debug statement either? Probably add a line in the DeleteFilter Constructor instead?

Reply:
Removed.